fearless/Modules/Staking/ControllerAccount/ControllerAccountInteractor.swift

```diff
@@ -0,0 +1,111 @@
+
+    init(
+        singleValueProviderFactory: SingleValueProviderFactoryProtocol,
+        substrateProviderFactory: SubstrateDataProviderFactoryProtocol,
+        runtimeService: RuntimeCodingServiceProtocol,
+        selectedAccountAddress: AccountAddress,
+        accountRepository: AnyDataProviderRepository<AccountItem>,
+        operationManager: OperationManagerProtocol,
+        feeProxy: ExtrinsicFeeProxyProtocol,
+        extrinsicService: ExtrinsicServiceProtocol
+    ) {
+        self.singleValueProviderFactory = singleValueProviderFactory
+        self.substrateProviderFactory = substrateProviderFactory
+        self.runtimeService = runtimeService
+        self.selectedAccountAddress = selectedAccountAddress
+        self.accountRepository = accountRepository
+        self.operationManager = operationManager
+        self.feeProxy = feeProxy
+        self.extrinsicService = extrinsicService
+    }
+
+    private func fetchAccounts() {
+        let operation = accountRepository.fetchAllOperation(with: RepositoryFetchOptions())
+        operation.completionBlock = { [weak self] in
+            DispatchQueue.main.async {
+                do {
+                    let accounts = try operation.extractNoCancellableResultData()
+                    self?.presenter.didReceiveAccounts(result: .success(accounts))
+                } catch {
+                    self?.presenter.didReceiveAccounts(result: .failure(error))
+                }
+            }
+        }
+
+        operationManager.enqueue(operations: [operation], in: .transient)
+    }
+}
+
+extension ControllerAccountInteractor: ControllerAccountInteractorInputProtocol {
+    func setup() {
+        stashItemProvider = subscribeToStashItemProvider(for: selectedAccountAddress)
+        fetchAccounts()
+        feeProxy.delegate = self
+    }
+
+    func estimateFee(controllerAddress: AccountAddress) {
+        do {
+            let setController = try callFactory.setController(controllerAddress)
+            let identifier = setController.callName + controllerAddress
+
+            feeProxy.estimateFee(using: extrinsicService, reuseIdentifier: identifier) { builder in
+                try builder.adding(call: setController)
+            }
+        } catch {
+            presenter.didReceiveFee(result: .failure(error))
+        }
+    }
+}
+
+extension ControllerAccountInteractor: SubstrateProviderSubscriber, SubstrateProviderSubscriptionHandler,
+    SingleValueProviderSubscriber, SingleValueSubscriptionHandler, AnyProviderAutoCleaning {
+    func handleStashItem(result: Result<StashItem?, Error>) {
+        do {
+            let maybeStashItem = try result.get()
+
+            clear(dataProvider: &accountInfoProvider)
+            presenter.didReceiveStashItem(result: result)
+
+            if let stashItem = maybeStashItem {
+                accountInfoProvider = subscribeToAccountInfoProvider(
+                    for: stashItem.controller,
```

Comment:
I think we need stash balance here because this account should send the extrinsic